From 8a2326fc33aabf16e8bbad63bc562220a1f42800 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 25 Jun 2018 15:14:04 +0200
Subject: [PATCH 077/137] pci: dwc: ipipe: enable interrupt pipelining

Fix up Synopsys's PCIE driver in order to channel interrupts through
the interrupt pipeline.
---
 drivers/pci/dwc/pcie-designware-host.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/dwc/pcie-designware-host.c b/drivers/pci/dwc/pcie-designware-host.c
index bc3e2d8..a5b8dd0 100644
--- a/drivers/pci/dwc/pcie-designware-host.c
+++ b/drivers/pci/dwc/pcie-designware-host.c
@@ -51,6 +51,7 @@ static int dw_pcie_wr_own_conf(struct pcie_port *pp, int where, int size,
 	.irq_disable = pci_msi_mask_irq,
 	.irq_mask = pci_msi_mask_irq,
 	.irq_unmask = pci_msi_unmask_irq,
+	.flags = IRQCHIP_PIPELINE_SAFE,
 };
 
 /* MSI int handler */
-- 
1.9.1

